viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
-# ------------- WAVE 1 --------------------
 
+# ------------- WAVE 1 --------------------
+# Creates a dictionary that holds movies with title, genre, and rating
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {"title": title,
+                 "genre": genre,
+                 "rating": rating}
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        return new_movie
+
+# Creates a dictionary for watched movies
+
+
+def add_to_watched(user_data, movie):
+    watched_move_list = {"watched": [movie]}
+    return watched_move_list
+
+# Creates a dictionary for movies in watchlist
+
+
+def add_to_watchlist(user_data, movie):
+    watchlist_list = {"watchlist": [movie]}
+    return watchlist_list
+
+# Removing watched movie from watchlist to watched within a dictionary
+
+
+def watch_movie(user_data, movie):
+    user_updated_data = user_data
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == movie:
+            remove_movie = user_updated_data["watchlist"].pop(i)
+            user_updated_data["watched"] += [remove_movie]
+    return user_updated_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# Calculates average rating for watched movies
+def get_watched_avg_rating(user_data):
+    counter = 0.0
+    for key, value in user_data.items():
+        for information in value:
+            counter += information["rating"]

```

Comment:
Since we know the key we want to look into already, let's skip the first loop completely. There might be only one key in _this_ particular dictionary, but we can't always assume. So let's go right to `user_data["watched"]`

```suggestion
    for movie in user_data["watched"]:
        counter += movie["rating"]

```